examples/braket_features/IonQ_Native_Gates.ipynb

```diff
@@ -0,0 +1,1661 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "87c15796-c783-4b91-bc28-4389df9f8308",
+   "metadata": {},
+   "source": [
+    "# IonQ Native Gates\n",
+    "\n",
+    "In this notebook, we will dive into the details of IonQ's native gates and their functionalities, enabling us to realize direct control over the quantum operations on the computer without compiler optimizations or error mitigation. We will discuss the native gates available on IonQ, their mathematical representations, and how they can be used for applications such as the quantum Fourier transform (QFT). \n",
+    "\n",
+    "IonQ's native gateset includes the GPi, GPi2, and arbitrary angle MS gate. These gates are physically implemented on IonQ hardware by utilizing resonant lasers to address ions through stimulated Raman transitions. The three gates provide a universal gate set for quantum computation. This means any quantum circuit can be realized with only these elementary operations. \n",
+    "\n",
+    "## Table of Contents\n",
+    "- Single-qubit gates\n",
+    "    - GPi gate: $X, Y$ gates\n",
+    "    - GPi2 gate: $R_X(\\pi/2), R_Y(\\pi/2)$ gates\n",
+    "    - General gate decompositions\n",
+    "    - Common single-qubit gates: $H$, $S$, $T$.\n",
+    "    - Writing native IonQ circuits in Braket \n",
+    "- Two-qubit gates: Arbitrary angle Mølmer-Sørenson gate\n",
+    "  - Mapping CNot into $XX$\n",
+    "  - Mapping controlled roots of Pauli gates into $XX$ gates\n",
+    "- Quantum Fourier transform with native gates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "266ddd70-259f-4b7b-8e6e-8d82165e3b4d",
+   "metadata": {},
+   "source": [
+    "## Single-qubit native gates\n",
+    "\n",
+    "### GPi gate\n",
+    "\n",
+    "The GPi gate is a fundamental single-qubit gate used in IonQ's trapped-ion quantum computing systems. It performs a rotation of $\\pi$ radians (180 degrees) on the Bloch sphere. What makes the GPi gate unique is its ability to rotate along any axis in the $xy$-plane of the Bloch sphere by adjusting the parameter $\\phi$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "0b7acc99-81bc-4bde-a1f1-6b84db2f3297",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The GPi gate is defined as:\n"
+     ]
+    },
+    {
+     "data": {
+      "text/latex": [
+       "$\\displaystyle \\left[\\begin{matrix}0 & e^{- i \\phi}\\\\e^{i \\phi} & 0\\end{matrix}\\right]$"
+      ],
+      "text/plain": [
+       "Matrix([\n",
+       "[         0, exp(-I*phi)],\n",
+       "[exp(I*phi),           0]])"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from sympy import I, Matrix, exp, pi, sqrt, symbols\n",
```

Comment:
Can you confirm if this `sympy` dependency needs to be pip installed [here](https://github.com/aws/amazon-braket-examples/blob/main/environment.yml)?

Reply:
If it is installed via BDK, we should be good.